e2e/interchain_accounts_test.go

```diff
@@ -0,0 +1,220 @@
+package e2e
+
+import (
+	"context"
+	"testing"
+
+	ibctest "github.com/strangelove-ventures/ibctest"
+	"github.com/strangelove-ventures/ibctest/chain/cosmos"
+	"github.com/strangelove-ventures/ibctest/ibc"
+	"github.com/strangelove-ventures/ibctest/test"
+	"github.com/stretchr/testify/suite"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	intertxtypes "github.com/cosmos/interchain-accounts/x/inter-tx/types"
+
+	"github.com/cosmos/ibc-go/e2e/testconfig"
+	"github.com/cosmos/ibc-go/e2e/testsuite"
+	"github.com/cosmos/ibc-go/e2e/testvalues"
+)
+
+func TestInterchainAccountsTestSuite(t *testing.T) {
+	// NOTE: this is a temporary mechanism to enable this test to run alongside the simd tests.
+	// This will be removed in a follow up PR and properly parameterized in a github workflow.
+	testconfig.SetChainBinaryVersions(
+		"ghcr.io/cosmos/ibc-go-icad", "master", "icad", "ghcr.io/cosmos/ibc-go-icad", "master",
+	)
+	suite.Run(t, new(InterchainAccountsTestSuite))
+}
+
+type InterchainAccountsTestSuite struct {
+	testsuite.E2ETestSuite
+}
+
+// RegisterICA will attempt to register an interchain account on the counterparty chain.
+func (s *InterchainAccountsTestSuite) RegisterICA(ctx context.Context, chain *cosmos.CosmosChain, user *ibctest.User, fromAddress, connectionID string) error {
```

Comment:
I think I'd prefer `RegisterInterchainAccount` to be explicit.

We could refactor this function to take the `MsgRegisterAccount` as an arg in place of the `fromAddr` and `connnectionID`. Then we can built the message with an explicit version for both incentivized and non-incentivized channels.

Reply:
good idea makes sense